builtin/providers/alicloud/resource_alicloud_nat_gateway.go

```diff
@@ -0,0 +1,308 @@
+package alicloud
+
+import (
+	"fmt"
+
+	"github.com/denverdino/aliyungo/common"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+	"strings"
+	"time"
+)
+
+func resourceAliyunNatGateway() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAliyunNatGatewayCreate,
+		Read:   resourceAliyunNatGatewayRead,
+		Update: resourceAliyunNatGatewayUpdate,
+		Delete: resourceAliyunNatGatewayDelete,
+
+		Schema: map[string]*schema.Schema{
+			"vpc_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"spec": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"bandwidth_package_ids": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"bandwidth_packages": &schema.Schema{
+				Type: schema.TypeList,
```

Comment:
As I see only 4 `bandwidth_packages` are allowed, we can add:

`MaxItems: 4` to the schema

We can then remove the if len() > 4 check below